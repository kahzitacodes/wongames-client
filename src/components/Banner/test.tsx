import { screen } from '@testing-library/react'

import { Banner } from '.'
import { renderWithTheme } from '@/utils/tests/helpers'

const props = {
  title: 'My title',
  subTitle: 'The subtitle',
  buttonLabel: 'See details',
  buttonLink: '/details',
  img: 'https://unsplash.com/pt-br/fotografias/uma-montanha-coberta-de-neve-e-nuvens-a-noite-Mud9nLcOwZY'
}

describe('<Banner />', () => {
  it('should render Banner correctly', () => {
    const { container } = renderWithTheme(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /my title/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/the subtitle/i)).toBeInTheDocument()
    expect(screen.getByRole('img', { name: /my title/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a ribbon', () => {
    renderWithTheme(
      <Banner {...props} ribbon="My ribbon" ribbonSize="sm" ribbonColor="red" />
    )

    const ribbon = screen.getByText(/my ribbon/i)

    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveStyle({
      backgroundColor: '#FB6E78'
    })

    expect(ribbon).toHaveStyle({
      height: '2.8rem',
      'font-size': '1.2rem'
    })
  })
})
