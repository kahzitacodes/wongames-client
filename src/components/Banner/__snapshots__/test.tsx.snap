// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render Banner correctly 1`] = `
.c5 {
  background: linear-gradient( 180deg,#F366A7 0%,#DC448A 40% );
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  font-weight: 600;
  line-height: 0%;
  display: flex;
  align-items: center;
  cursor: pointer;
  gap: 0.8rem;
  background-size: 200% 200%;
  transition: background-position 0.2s linear;
  text-decoration: none;
  height: 4.8rem;
  font-size: 1.6rem;
  padding: 0 2.4rem;
  width: fit-content;
}

.c5:hover {
  background-position: 0 100%;
}

.c5 svg {
  width: 2.4rem;
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 24rem;
  background-color: #ECECEF;
  background-image: url(https://unsplash.com/pt-br/fotografias/uma-montanha-coberta-de-neve-e-nuvens-a-noite-Mud9nLcOwZY);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: #2C2E3E;
  padding: 1.6rem 2.4rem 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  margin-bottom: 1.6rem;
}

.c4 strong {
  color: #2496FF;
  font-weight: 600;
}

@media (max-width: 1170px) {

}

@media (min-width: 768px) {
  .c0 {
    box-shadow: 0 0.4rem 1rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width: 768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width: 768px) {
  .c2 {
    background-color: rgba(5, 6, 26, 0.7);
    borderRadius: 0.8rem;
    padding: 3.2rem 4.0rem 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width: 768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width: 768px) {
  .c4 {
    font-size: 1.8rem;
    margin-bottom: 3.2rem;
  }
}

<div
  class="c0"
>
  <div
    aria-label="My title"
    class="c1"
    role="img"
    src="https://unsplash.com/pt-br/fotografias/uma-montanha-coberta-de-neve-e-nuvens-a-noite-Mud9nLcOwZY"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      My title
    </h2>
    <p
      class="c4"
    >
      The subtitle
    </p>
    <a
      class="c5"
      href="/details"
    >
      <span>
        See details
      </span>
    </a>
  </div>
</div>
`;
